### PR TITLE
Store `is_liquidity_order` in the database

### LINKED ISSUE #176: Store `is_liquidity_order` in the database
Currently detecting liquidity orders placed through our API happens by looking up the owner of the order in an allow-list of private market makers.
This list can be configured with `--liquidity-order-owners addr1,addr2,addr3`.
If the owner of an order is currently in that list it gets treated as a liquidity order regardless of whether it was a liquidity order when we it got stored in the database.

The proper approach would be to decide whether an order is a liquidity order before storing it in the database and storing that information along with it. That would prevent an order changing its type after reconfiguring and restarting the service.

I think there are 2 ways this could be implemented:
1. Keep most things as they are right now and add the field `is_liquidity_order` to `OrderMetadata` before storing it in the DB.
2. Add `is_liquidity_order` to `OrderCreation`. This would allow users to decide whether they want a liquidity order or a normal order. It would still make sense to lock that feature behind an allow-list, though. To not break compatibilty of the `/create_order` endpoint the field `is_liquidity_order` could be optional with `false` as the default.

### PR DESCRIPTION
**Pull request by MartinquaXD**
Fixes #176

Whether an order is a liquidity order currently depends on the `--liquidity-order-owners` argument. If the owner of an order is in that list the order is considered a liquidity order. This is a bit awkward because the arguments for the order book and driver might get out of sync. This could cause the order book to consider an order a liquidity order whereas the driver considers it a user order.
To make the type of an order consistent across restarts of the order book its value needs to get stored in the database.

Eventually we should move away from allow-listing individual addresses and extending the API such that each user can decide whether to create a liquidity order or a normal order.
To not make a gigantic PR while breaking the current behavior (deciding for the user whether to create a normal or liquidity order) this PR implements that change somehow awkwardly.

Things I want to do which are outside of the scope of this PR:
Drop `--liquidity-order-owners` argument entirely.
Add `is_liquidity_order` to the `OrderCreationPayload` to allow users to decide what order type they want to create.

### Test Plan
Updated unit tests
Manual test of the migration script verifying that existing partially fillable orders get marked as liquidity orders and that no default value for `is_liquidity_order` remains.
Manual test verifying that the content of `isLiquidityOrder` in the auction endpoint does not change after restarting the order book with a new list of `--liquidity-order-owners`. This was done in both directions:
1. Start with allow-list and verify that order is still liquidity order after restarting without list
2. Start without allow-list and verify that order is still normal order after restarting with the list

### CODE REVIEW

File: database/sql/V020__extend_orders_with_is_liquidity_order.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+ALTER TABLE orders ADD is_liquidity_order boolean NOT NULL DEFAULT false;
+-- So far every liquidity order was also a partially fillable order and all
```
Comment (nlordell) [inline database/sql/V020__extend_orders_with_is_liquidity_order.sql:2]: I think this comment may be wrong.

Specifically, we used to allow liquidity orders from stable coin bots with fill or kill orders before introducing partially fillable order support. As such, while all partially fillable orders are liquidity orders, not all liquidity orders are partially fillable orders.

I would reword this to indicate that this provides us with a best guess of what orders were partially fillable, and that other orders are lost to time.

**Review by nlordell — APPROVED** [review]
LGTM! Only issue is that the comment in the migration script is slightly wrong.

Approving assuming a resolution to the comment and that the TODO for testing migration script is done.

File: database/sql/V020__extend_orders_with_is_liquidity_order.sql [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+-- partially fillable orders were liquidity orders.
+-- We use this fact to initialize is_liquidity_order for existing orders correctly.
+UPDATE orders SET is_liquidity_order = partially_fillable WHERE is_liquidity_order = false;
```
Comment (vkgnosis) [inline database/sql/V020__extend_orders_with_is_liquidity_order.sql:5]: Looked through the Postgres docs a bit. I think we want
```
ALTER TABLE orders  ALTER COLUMN is_liquidity_order DROP DEFAULT;
```
at the end of the migration. Otherwise we will have set the false default set forevery which will make Postgres use it for all future insert operations that don't specify the column. We don't want this behavior we just need the default temporarily before the new value is calculated in the second command.

The second command could also be simplified to this I believe
```
UPDATE orders SET is_liquidity_order = partially_fillable
```

**Review by sunce86 — APPROVED** [review]

**Review by vkgnosis — APPROVED** [review]
